fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -655,8 +667,18 @@ def state_dict(self, *args: Any, **kwargs: Any) -> "OrderedDict[str, torch.Tenso
             self._cast_buffers()
         return state_dict
 
-    # TODO (Min): figuring out how to do typing for this overloaded function.
-    def local_state_dict(self, *args, **kwargs):  # type: ignore
+    @typing.overload
+    def local_state_dict(
+        self, destination: Mapping[str, torch.Tensor], prefix: str = ..., keep_vars: bool = ...
+    ) -> Mapping[str, torch.Tensor]:
+        ...
+
+    @typing.overload
+    def local_state_dict(self, prefix: str = ..., keep_vars: bool = ...) -> "OrderedDict[str, torch.Tensor]":
+        ...
```

Comment:
Are you doing this all to please mypy? We need both overloads, I assume?

Reply:
yeah, it is *the official way* to do overloaded function as far as I can tell. if we only have the (any, any)->any signature, it doesn't match the parent class signature and I suppose this will help typing checking our code a tiny bit better.